pkg/reconciler/security/istio/eventpolicybinding/eventpolicybinding.go

```diff
@@ -82,15 +86,20 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, b *v1alpha1.EventPolicyB
 		return err
 	}
 
-	// Copy the HTTP policy binding status.
-	b.Status = *hpb.Status.DeepCopy()
+	// Propagate the status.
+	if hpb.Status.IsReady() {
+		b.Status.MarkBindingAvailable()
+	} else {
+		st := hpb.Status.GetTopLevelCondition()
+		b.Status.MarkBindingFailure(st.Reason, st.Message)
+	}
```

Comment:
nit: might be cleaner to just create a `b.Status.PropagateBindingStatus(hpb)`, and you put this logic there. 
That is what we've been doing for propagating deployments statuses, and other objects as well.
See for example: https://github.com/knative/eventing/blob/1c825f9a6478582a9f803e445d1989272515421c/pkg/reconciler/apiserversource/apiserversource.go#L138